PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/serviceexplorer/azure/appservice/ProfileFlightRecordAction.java

```diff
@@ -0,0 +1,202 @@
+ * SOFTWARE.
+ */
+
+package com.microsoft.intellij.serviceexplorer.azure.appservice;
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.application.ModalityState;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.microsoft.azure.management.appservice.OperatingSystem;
+import com.microsoft.azure.management.appservice.WebAppBase;
+import com.microsoft.azure.toolkit.intellij.appservice.jfr.RunFlightRecorderDialog;
+import com.microsoft.azure.toolkit.lib.appservice.jfr.FlightRecorderConfiguration;
+import com.microsoft.azure.toolkit.lib.appservice.jfr.FlightRecorderManager;
+import com.microsoft.azure.toolkit.lib.appservice.jfr.FlightRecorderStarterBase;
+import com.microsoft.intellij.util.PluginUtil;
+import com.microsoft.tooling.msservices.components.DefaultLoader;
+import com.microsoft.tooling.msservices.helpers.Name;
+import com.microsoft.tooling.msservices.serviceexplorer.NodeActionEvent;
+import com.microsoft.tooling.msservices.serviceexplorer.NodeActionListener;
+import com.microsoft.tooling.msservices.serviceexplorer.azure.webapp.WebAppNode;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.StringUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.concurrent.CountDownLatch;
+import java.util.logging.Logger;
+
+@Name("Profile Flight Recorder")
+public class ProfileFlightRecordAction extends NodeActionListener {
+    private static final Logger logger = Logger.getLogger(ProfileFlightRecordAction.class.getName());
+    private static final String PROFILE_FLIGHT_RECORDER = "Profile Flight Recorder";
+    private static final int ONE_SECOND = 1000;
+    private static final int TWO_SECONDS = 2000;
+    private final Project project;
+    private final WebAppBase appService;
+
+    public ProfileFlightRecordAction(WebAppNode webAppNode) {
+        super();
+        this.project = (Project) webAppNode.getProject();
+        appService = webAppNode.getWebapp();
+    }
+
+    @Override
+    protected void actionPerformed(NodeActionEvent nodeActionEvent) {
+        // prerequisite check
+        if (appService.operatingSystem() == OperatingSystem.LINUX &&
+                StringUtils.containsIgnoreCase(appService.linuxFxVersion(), "DOCKER|")) {
+            notifyUserWithErrorMessage("Flight Record not supported", String.format("Docker app service '%s' is not "
+                                                                                     + "supported.",
+                                                        appService.name()));
+            return;
+        }
+        if (StringUtils.equalsIgnoreCase(appService.state(), "running")) {
+            notifyUserWithErrorMessage("App service not running", String.format("App service '%s' is not running.",
```

Comment:
We may prompt user and ask whether to start app service and then run flight record

Reply:
trivial difference